### ISSUE TITLE
invalid application of 'sizeof' to an incomplete type 'const void'

### ISSUE THREAD

**Issue by Arcnor** [bytedeco/javacpp#189]
From https://groups.google.com/forum/#!topic/javacpp-project/mYpan04nTUw

For the example class:

```cpp
#include <functional>

class Test {
	public:
		Test(const std::function<void(bool)> &callback = std::function<void(bool)>());
		void setCallback(const std::function<void(bool)> &callback) { mCallback = callback; }
	protected:
		std::function<void(bool)> mCallback;
};
```

This will generate:

```java
// Parsed from test.h

// #include <functional>

@NoOffset public static class Test extends Pointer {
    static { Loader.load(); }
    /** Pointer cast constructor. Invokes {@link Pointer#Pointer(Pointer)}. */
    public Test(Pointer p) { super(p); }
    /** Native array allocator. Access with {@link Pointer#position(long)}. */
    public Test(long size) { super((Pointer)null); allocateArray(size); }
    private native void allocateArray(long size);
    @Override public Test position(long position) {
        return (Test)super.position(position);
    }

		public Test(@Const @ByRef(nullValue = "std::function<void(bool)>()") FuncBool callback) { super((Pointer)null); allocate(callback); }
		private native void allocate(@Const @ByRef(nullValue = "std::function<void(bool)>()") FuncBool callback);
		public Test() { super((Pointer)null); allocate(); }
		private native void allocate();
		public native void setCallback(@Const @ByRef FuncBool callback);
}
```

On the `InfoMap` we define:

```java
infoMap.put(new Info("std::function<void(bool)>").pointerTypes("FuncBool"));
```

`FuncBool` looks like:

```java
@Opaque
@NoOffset
public static class FuncBool extends FunctionPointer {
	static { Loader.load(); }
	/** Pointer cast constructor. Invokes {@link Pointer#Pointer(Pointer)}. */
	public FuncBool(Pointer p) { super(p); }
	protected FuncBool() { allocate(); }
	private native void allocate();
	public native void call(@Cast("bool") boolean value);
}
```

...but when trying to compile the generated C++ code, an error will be thrown. The solution seems to be to remove the `@Const` from the `setCallback` and constructor callback parameters.

**Comment by saudet** [comment]
Does this happen only with `std::function` and `FunctionPointer`?

**Comment by Arcnor** [comment]
So far, this is the only time I've encountered something like that, yeah.

**Comment by saudet** [comment]
Strange, I can't reproduce that here. Everything builds fine with the test code above and either GCC 6.3.1 or Clang 3.9.1.

**Comment by Arcnor** [comment]
Strange indeed. I'll share my project with you later, and you'll probably see the issue there.

**Comment by saudet** [comment]
Have you been able to diagnose this? or create a small self-contained example to reproduce it?

**Comment by Arcnor** [comment]
No, ended up writing JNI manually for now, which was a pain but I had a deadline. I'll comment here if I take this project again or find a way of reproducing it (I think I tried with a smaller version but couldn't, although can't remember properly)

**Comment by saudet** [comment]
Ok, sorry I couldn't be of any help :(

**Comment by Arcnor** [comment]
You're very helpful every single time I ask, so please don't be sorry :). If I ever need this again (which I might) I'll try to track the problem down again.

Thanks!

**Comment by saudet** [comment]
BTW, when there are parts of header files that we don't really care about, it's always possible to skip those parts, as with LAPACK here:
https://github.com/bytedeco/javacpp-presets/blob/master/openblas/src/main/java/org/bytedeco/javacpp/presets/openblas.java#L55

**Comment by saudet** [comment]
This issue has probably been fixed with commit c03cd9d335820ba9a26bf200fe90a69400001303, but please let me know if you ever encounter it again, thanks!
